crml/governance/src/lib.rs

```diff
@@ -240,25 +290,50 @@ decl_module! {
 			ProposalStatus::insert(proposal_id, ProposalStatusInfo::ApprovedEnactmentCancelled);
 			ProposalCalls::remove(proposal_id);
 			ProposalVotes::remove(proposal_id);
+			<ReferendumStartTime<T>>::remove(proposal_id);
 
 			Ok(())
 		}
 
+		/// Submit a veto for a referendum
+		#[weight = 1_000_000]
+		fn vote_against_referendum(
+			origin,
+			proposal_id: ProposalId,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			ensure!(Self::proposal_status(proposal_id) == Some(ProposalStatusInfo::ReferendumDeliberation), Error::<T>::ReferendumEnded);
```

Comment:
not sure this is the right status to mark ended,
did you mean these?

ProposalStatusInfo::ApprovedWaitingEnactment
ProposalStatusInfo::ApprovedEnacted

Reply:
It is meant to be ReferendumDeliberation but the error message is unclear, have fixed it

Reply:
ah I read this backward, it's checking its in the deliberation stage 👍